version-compatibility/0.14.1-0.14.2/fuel-core:0.14.1-client:0.14.2/test.rs

```diff
@@ -0,0 +1,56 @@
+#[cfg(test)]
+mod test {
+    use fuel_core::{
+        database::Database,
+        service::{
+            Config,
+            FuelService,
+        },
+    };
+    use fuel_gql_client::{
+        client::{
+            types::TransactionStatus,
+            FuelClient,
+        },
+        fuel_tx,
+        fuel_tx::UniqueIdentifier,
+    };
+
+    #[tokio::test]
+    async fn produce_block_compatible() {
+        let db = Database::default();
+
+        let mut config = Config::local_node();
+
+        config.manual_blocks_enabled = true;
+
+        let srv = FuelService::from_database(db, config.clone())
+            .await
+            .unwrap();
+
+        let client = FuelClient::from(srv.bound_address);
+
+        let new_height = client.produce_blocks(5, None).await.unwrap();
+
+        assert_eq!(5, new_height);
+
+        let tx = fuel_tx::Transaction::default();
+        client.submit_and_await_commit(&tx).await.unwrap();
+
+        let transaction_response = client
+            .transaction(&format!("{:#x}", tx.id()))
+            .await
+            .unwrap();
+
+        if let TransactionStatus::Success { block_id, .. } =
+            transaction_response.unwrap().status
+        {
+            // TODO: We need to decide is that breaking change or not. For now, the test fails,
```

Comment:
If `fuel-client` is `0.14.2` but `fuel-core` is `0.14.1` the test fails with "Unknown field 'consensus'". 
We have 4 variants on how to fix that=)

1. Make a breaking change `0.15.0`.
1. Extract `consensus` from `Block` into a separate method and on the client side try to fetch that part if it is possible in a separate request.
1. Do two requests, one with a new version of the block, if it fails, send it with the old version=)
1. Add a new method to the `Block` in `fuel-client` to get the `consensus` field by passing `FuelClient`=)

Reply:
Option #1 would be the cleanest, as fuel-client 0.15.0 would still be compatible with fuel-core 0.14.1.